### PR TITLE
Implement Retrieve Product Identifier API

### PR DESCRIPTION
**Pull request by cb-haripriyan**
Recommended to review these in order:
https://github.com/chargebee/chargebee-react-native/pull/56
https://github.com/chargebee/chargebee-react-native/pull/57
https://github.com/chargebee/chargebee-react-native/pull/58
- Implements the retrieve product identifiers API

### CODE REVIEW

File: example/src/App.tsx [hunk 1]
```diff
@@ -1,34 +1,51 @@
...
+        onPress={retrieveProductIdentifiers}
+      />
     </View>
   );
 }
 
+const retrieveProductIdentifiers = () => {
+  const queryParams = new Map<string, string>();
+  queryParams.set('limit', '100');
+  Chargebee.retrieveProductIdentifiers(queryParams)
```
Comment (cb-sriramthiagarajan) [inline example/src/App.tsx:32]: Can we use `async/await` here (and throughout) instead of `.then` for better readability?

Comment (cb-haripriyan) [reply example/src/App.tsx]: @cb-sriramthiagarajan Updated this.

File: android/src/main/java/com/chargebee/android/utils/Converter.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.chargebee.android.utils
+
+import com.facebook.react.bridge.WritableArray
+import com.facebook.react.bridge.WritableNativeArray
+
+internal fun convertArrayToWritableArray(array: Array<Any?>): WritableArray {
```
Comment (cb-sriramthiagarajan) [inline android/src/main/java/com/chargebee/android/utils/Converter.kt:6]: If we're only looking for `String` (or `null`), should `array` be of type `Array<String>`? 🤔

**Review by cb-sriramthiagarajan — APPROVED** [review]

File: src/NativeChargebeeReactNative.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { TurboModuleRegistry } from 'react-native';
 
 export interface Spec extends TurboModule {
   configure(site: string, publishableApiKey: string, sdkKey: string): void;
+  retrieveProductIdentifiers(queryParams: Object): Promise<Array<string>>;
```
Comment (cb-sriramthiagarajan) [inline src/NativeChargebeeReactNative.ts:6]: Can we type `queryParams` more specifically?

Comment (cb-sriramthiagarajan) [reply src/NativeChargebeeReactNative.ts:6]: Also, should this be an optional param with default value? 🤔

Comment (cb-haripriyan) [reply src/NativeChargebeeReactNative.ts:6]: @cb-sriramthiagarajan I was following the listed argument types for conversion between JS and Kotlin/iOS byt following [this](https://reactnative.dev/docs/native-modules-android.html#argument-types) and [this](https://reactnative.dev/docs/native-modules-ios.html#argument-types)
For passing in queryParams which is passed in as key-value/dictionary, the recommended type seems to `Object`. 
Let me try if subtyping works.

File: android/src/main/java/com/chargebee/android/utils/Converter.kt (hunk 2 above)
Comment (cb-haripriyan) [reply android/src/main/java/com/chargebee/android/utils/Converter.kt:6]: Let me try this out.

Comment (cb-haripriyan) [reply android/src/main/java/com/chargebee/android/utils/Converter.kt:6]: This is done.

**Pull request merged**